tests/sanitizers/test_sanitizers.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+set -e
+# for info see 'read.me'
+
+DIR_BUILD=$PWD
+DIR_SRC=$(cd $(dirname "$0")/../..; pwd -P)
+DIR_REPORT="$DIR_BUILD/sanitizers/report_$(date +"%y-%m-%d-%H%M%S")"
+DIR_GTEST_BUILD="$DIR_BUILD/gtest"
+DIR_GTEST_INSTALL="$DIR_GTEST_BUILD/googletest/install"
+GTEST_SETUP="$DIR_SRC/ci/travis/setup_gtest.sh"
+REPORT_BUILD="$DIR_REPORT/report_build"
+REPORT_CTEST="$DIR_REPORT/ctest_log"
+
+# Set a default sanitizer if envvar is not set
+if [ -z "$SANITIZER" ]
+then
+    SANITIZER="ASAN"
+    echo "INFO: environment variable SANITIZER is not set. Will be using by default $SANITIZER"
+fi
+
+# ASAN: Address Sanitizer
+ASAN="-fsanitize=address "
+ASAN_FLAGS="-fsanitize-address-use-after-scope -fsanitize-recover=address -fno-omit-frame-pointer -fno-common "
+# LSAN: Leak Sanitizer
+LSAN="-fsanitize=leak "
+LSAN_FLAGS=""
+# UBSAN: UndefinedBehaviour Sanitizer
+UBSAN="-fsanitize=undefined "
+UBSAN_FLAGS="-fno-omit-frame-pointer "
+# Compile time out of bounds checks. See more: https://fuzzing-project.org/tutorial-cflags.html
+EXTRA_FLAGS="-Wformat -Werror=format-security -Werror=array-bounds "
+
+FLAGS=${SANITIZER}"_FLAGS"
+SANITIZER_LDFLAGS="${!SANITIZER}"
+SANITIZER_FLAGS="${!SANITIZER}${!FLAGS}"
+SANITIZER_FLAGS+="$EXTRA_FLAGS"
+SANITIZER_FLAGS+="-shared-libasan "
+# NOTE: option '-shared-libasan' for link shared library with:
+# ASAN as a shared library. See more: https://github.com/google/sanitizers/wiki/AddressSanitizerAsDso
+# UBSAN as a shared library. See more: https://reviews.llvm.org/D38525
+
+CXX_VERSION_OUTPUT=$(clang --version)
+CXX_VERSION=$(echo $CXX_VERSION_OUTPUT | cut -d"-" -f1 | cut -d" " -f4)
```

Comment:
This prints nothing
$ echo $CXX_VERSION



$ echo $CXX_VERSION_OUTPUT
clang version 9.0.0-2 (tags/RELEASE_900/final) Target: x86_64-pc-linux-gnu Thread model: posix InstalledDir: /usr/bin


try 
clang --version | head -n 1 | grep -o -E "[[:digit:]].[[:digit:]].[[:digit:]]" | uniq | sort

Reply:
The expression contain regex that have a mistake, but it is not a problem.
True way: is read source code (#define ...), but not an interpretation output for human.